core/commands/SubprojectAddCommand.php

```diff
@@ -0,0 +1,139 @@
+	 * - project_id (integer)
+	 *
+	 * $p_data['payload'] is expected to contain:
+	 * - subproject_id (integer)
+	 * - inherit_parent (bool)
+	 *
+	 * @param array $p_data The command data.
+	 */
+
+	/**
+	 * @var integer
+	 */
+	private $project_id;
+
+	/**
+	 * @var integer
+	 */
+	private $subproject_id;
+
+	/**
+	 * @var boolean
+	 */
+	private $inherit_parent;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {		
+		$this->project_id = helper_parse_id( $this->query( 'project_id' ), 'project_id' );
+
+		if( !access_has_project_level( config_get( 'manage_project_threshold' ), $this->project_id ) ) {
+			throw new ClientException(
+				'Access denied to add subprojects',
+				ERROR_ACCESS_DENIED );
+		}
+
+		if ( config_get( 'subprojects_enabled' ) == OFF ) {
+			throw new ClientException(
+				'Access denied to add subprojects',
+				ERROR_ACCESS_DENIED );
+		}
+
+		if( !project_exists( $this->project_id )) {
+			throw new ClientException(
+				"Project \"$this->project_id\"not found",
+				ERROR_PROJECT_NOT_FOUND,
+				array( $this->project_id ) );
+		}
+
+		$this->subproject_id = helper_parse_id( $this->payload( 'subproject_id' ), 'subproject_id' );
+		if( !project_exists( $this->subproject_id )) {
+			throw new ClientException(
+				"Project \"$this->subproject_id\" not found",
+				ERROR_PROJECT_NOT_FOUND,
+				array( $this->subproject_id ) );
+		}
+
+		if( $this->project_id == $this->subproject_id ) {
+			throw new ClientException(
+				"Project can't be subproject of itself",
+				ERROR_PROJECT_RECURSIVE_HIERARCHY );
+		}
+		
+		if( in_array( $this->subproject_id, project_hierarchy_get_subprojects( $this->project_id, true ) ) ) {
+			throw new ClientException(
+				"Project \"$this->subproject_id\" is already a subproject of \"$this->project_id\"",
+				ERROR_PROJECT_SUBPROJECT_DUPLICATE,
+				array( $this->subproject_id, $this->project_id ) );
+		}		
+
+		if( in_array( $this->project_id, project_hierarchy_get_all_subprojects( $this->subproject_id, true ) ) ) {
+			throw new ClientException(
+				"Project can't be a descendant subproject of itself",
+				ERROR_PROJECT_RECURSIVE_HIERARCHY );
+		}		
+
+		$this->inherit_parent = $this->payload( 'inherit_parent', true );
+	}
+
+	/**
+	 * Process the command.
+	 *
+	 * @returns array Command response
+	 */
+	protected function process() {
+		if( $this->project_id != helper_get_current_project() ) {
+			# in case the current project is not the same project of the bug we are
+			# viewing, override the current project. This to avoid problems with
+			# categories and handlers lists etc.
+			global $g_project_override;
+			$g_project_override = $this->project_id;
+		}
+
+		project_hierarchy_add( $this->subproject_id, $this->project_id , $this->inherit_parent );
+
+		return array( 'parent_id' => $this->project_id,
```

Comment:
Not sure if this is needed, but if it is then we should return "project": { "id": 1234, "name ": "xyz" }.  I believe there is a helper method that returns a project reference structure which contains these two fields.